net/libreswan/files/default.hotplug

```diff
@@ -0,0 +1,220 @@
+#
+#	PLUTO_CFG_CLIENT=0|1
+#		is MODECFG or IKEv2 Config client.
+#
+#	PLUTO_CFG_SERVER=0|1
+#		is MODECFG or IKEv2 Config server.
+#
+#	PLUTO_PEER_DNS_INFO
+#		The peer's supplied DNS information (IKEv1 and IKEv2)
+#
+#	PLUTO_PEER_DOMAIN_INFO
+#		The peer's supplied domain list for local resolving (IKEv2 only)
+#
+#	PLUTO_PEER_BANNER
+#		is the peer's provided banner
+#
+#	PLUTO_NM_CONFIGURED=0|1
+#		is NetworkManager used for resolv.conf update
+#
+#	PLUTO_CONN_ADDRFAMILY
+#		is the family type, "ipv4" or "ipv6"
+#
+#	PLUTO_CONN_KIND
+#		is the "kind" of connection (CK_PERMANENT, CK_INSTANCE, etc)
+#
+#	PLUTO_STACK
+#		is the local IPsec kernel stack used, eg XFRM, BSDKAME, NOSTACK
+#
+#	PLUTO_IS_PEER_CISCO=0|1
+#		remote server type is cisco. Add support for cisco extensions
+#		when used with xauth.
+#
+#	PLUTO_SA_REQID
+#		When using KAME or XFRM, the IPsec SA reqid base value.
+#		ESP/AH out is base, ESP/AH in = base + 1
+#		IPCOMP is base + 2 plus for inbound + 1
+#
+#	PLUTO_XFRMI_FWMARK
+#		use outgoing mark
+#
+#	PLUTO_SA_TYPE
+#		The type of IPsec SA (ESP or AH)
+#
+#	PLUTO_USERNAME
+#		The username (XAUTH or GSSAPI) that was authenticated (if any)
+#		for this SA
+#
+# 	PLUTO_VIRT_INTERFACE
+#		is the name of ipsec interface used by clear traffic in/out
+#
+# 	INTERFACE_IP
+# 		The IP to configure / expect on the interface? Currently is never set
+#
+#	PLUTO_XFRM_ROUTE
+#		if an XFRM (ipsec-device) has been specified, value will be "yes"
+#
+#	XAUTH_FAILED
+#		If xauthfail=soft this will be set to 1 if XAUTH authentication
+#		failed. If xauthfail=hard, the updown scripts never run.
+#
+#	CONNMARK
+#		If mark= is set on the connection, this variable will be
+#		set with the value. It can be used for iptables or VTI.
+#
+#	CONNMARK_IN
+#		the incoming mark to use
+#
+#	CONNMARK_OUT
+#		the outgoing mark to use
+#
+#	VTI_IFAC=iface
+#		Name of VTI interface to create
+#
+#	VTI_ROUTING=yes|no
+#		Whether or not to perform ip rule and ip route commands
+#		covering the IPsec SA address ranges to route those packets
+#		into the VTI_IFACE interface. This should be enabled unless
+#		the IPsec SA covers 0.0.0.0/0 <-> 0.0.0.0/0
+#
+#	VTI_SHARED=yes|no
+#		Whether or not more conns (or instances) share a VTI device.
+#               If not shared, the VTI device is deleted when tunnel goes down.
+#
+#	VTI_IP
+#		The IP to configure on the VTI device
+#
+#	SPI_IN / SPI_OUT
+#		The inbound and outbound SPI's of the connection.
+#
+#	PLUTO_INBYTES
+#		total bytes received
+#
+#	PLUTO_OUTBYTES
+#		total bytes sent
+#
+#	NFLOG
+#		is the nflog group to use
+#
+
+case "${PLUTO_VERB}" in
+	prepare-host|prepare-host-v6) ;;
+	prepare-client|prepare-client-v6) ;;
+	route-host|route-host-v6) ;;
+	unroute-host|unroute-host-v6) ;;
+	route-client|route-client-v6) ;;
+	unroute-client|unroute-client-v6) ;;
+	up-host|up-host-v6) ;;
+	down-host|down-host-v6) ;;
+	up-client|up-client-v6) ;;
+	down-client|down-client-v6) ;;
```

Comment:
Shouldn't we call /etc/hotplug.d/libreswan?
Maybe there are other system components that want to be informed when something changes.
They would then simply place a script in this folder and it would then be executed.

Reply:
This file is installed under hotplug directory, It is kept there as a template only so it can be copied and used.

Reply:
This is then always called up and slows down the performance. I would not recommend to  install the script on the system. But leave it in the git repository so that others can use it as a template.

Reply:
Done